week_04/src/model/Book.java

```diff
@@ -0,0 +1,83 @@
+package model;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.Objects;
+import java.util.UUID;
+
+public class Book {
+    private String ISBN; // UUID unique identifier
+    private String title;
+    private String author;
+    private BigDecimal price;
+    private String publisher;
+    private LocalDate publishedYear;
+
+    public Book() {
+        this.ISBN = UUID.randomUUID().toString();
+    }
+
+    public Book(String title, String author, BigDecimal price, String publisher, LocalDate publishedYear) {
+        this.title = title;
```

Comment:
Keep in mind that if you use this constructor to create an Object then the default constructor wont be executed.
This means that you won't have ISBIN.